components/PercentageInput.tsx

```diff
@@ -1,28 +1,89 @@
 import React from "react";
+import { Range } from "react-range";
 
 interface PercentageInputProps {
   value: number;
   onChange: (newValue: number) => void;
 }
 
 function PercentageInput({ value, onChange }: PercentageInputProps) {
-  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    let newValue = parseInt(event.target.value, 10);
-
+  const handleChange = (number: number) => {
     // Ensure the value is between 0 and 100
-    newValue = Math.max(0, Math.min(newValue, 100));
+    const newValue = Math.max(0, Math.min(number, 100));
 
     onChange(newValue);
   };
 
   return (
-    <input
-      className="rounded-md px-4 py-2 bg-inherit border mb-6"
-      type="number"
-      value={value}
-      onChange={handleChange}
-      min={0}
+    <Range
+      step={50}
+      min={50}
       max={100}
+      values={[value]}
+      onChange={(values) => handleChange(values[0])}
+      renderTrack={({ props, children }) => (
+        <div
+          {...props}
+          style={{
+            ...props.style,
+            height: "6px",
+            width: "100%",
+            backgroundColor: "#ccc",
+          }}
+        >
+          {children}
+        </div>
+      )}
+      renderMark={({ props, index }) => (
```

Comment:
this function could be brought out and then you'd do `renderMark={renderMark}` with the `renderMark` definition above.